src/main/java/io/supertokens/emailpassword/PasswordHashing.java

```diff
@@ -113,32 +132,32 @@ public boolean verifyPasswordWithHash(String password, String hash) {
         if (PasswordHashingUtils.isInputHashInArgon2Format(hash)) {
             ProcessState.getInstance(main).addState(ProcessState.PROCESS_STATE.PASSWORD_VERIFY_ARGON, null);
             if (hash.startsWith("$argon2id")) {
-                return withConcurrencyLimited(() -> argon2id.verify(hash, password.toCharArray()));
+                return withArgon2HashingConcurrencyLimited(() -> argon2id.verify(hash, password.toCharArray()));
             }
 
             if (hash.startsWith("$argon2i")) {
-                return withConcurrencyLimited(() -> argon2i.verify(hash, password.toCharArray()));
+                return withArgon2HashingConcurrencyLimited(() -> argon2i.verify(hash, password.toCharArray()));
             }
 
-            if (hash.startsWith("$argon2d")) { // argon2 hash looks like
-                                               // $argon2id$v=..$m=..,t=..,p=..$tgSmiYOCjQ0im5U6...
-                return withConcurrencyLimited(() -> argon2d.verify(hash, password.toCharArray()));
+            if (hash.startsWith("$argon2d")) {
+                return withArgon2HashingConcurrencyLimited(() -> argon2d.verify(hash, password.toCharArray()));
             }
         } else if (PasswordHashingUtils.isInputHashInBcryptFormat(hash)) {
             ProcessState.getInstance(main).addState(ProcessState.PROCESS_STATE.PASSWORD_VERIFY_BCRYPT, null);
             String bCryptPasswordHash = PasswordHashingUtils
                     .replaceUnsupportedIdentifierForBcryptPasswordHashVerification(hash);
             return BCrypt.checkpw(password, bCryptPasswordHash);
+        } else if (PasswordHashingUtils.isInputHashInFirebaseSCryptFormat(hash)) {
+            ProcessState.getInstance(main).addState(ProcessState.PROCESS_STATE.PASSWORD_VERIFY_FIREBASE_SCRYPT, null);
+            return withFirebaseSCryptConcurrencyLimited(() -> PasswordHashingUtils
+                    .verifyFirebaseSCryptPasswordHash(password, hash, Config.getConfig(main).getFirebaseSigningKey()));
```

Comment:
you do not have a null check for the value returned by getFirebaseSigningKey. Nor is there a null check for that in the config class